src/error.rs

```diff
@@ -19,6 +19,16 @@ pub enum IrohError {
     Connection { description: String },
     #[error("blob: {description}")]
     Blob { description: String },
+    #[error("Ipv4Addr error: {description}")]
+    Ipv4Addr { description: String },
```

Comment:
not needed for this PR, but we probably should use a seperate error type for these conversion errors probably